ts-sdk/whirlpool/tests/increaseLiquidity.test.ts

```diff
@@ -1,5 +1,480 @@
+
+  beforeAll(async () => {
+    mintA = await setupMint();
+    mintB = await setupMint();
+    mintTEA = await setupMintTE();
+    mintTEB = await setupMintTE();
+    mintTEFee = await setupMintTEFee();
+
+    ataMap[mintA] = await setupAta(mintA, { amount: tokenBalance });
+    ataMap[mintB] = await setupAta(mintB, { amount: tokenBalance });
+    ataMap[mintTEA] = await setupAtaTE(mintTEA, { amount: tokenBalance });
+    ataMap[mintTEB] = await setupAtaTE(mintTEB, { amount: tokenBalance });
+    ataMap[mintTEFee] = await setupAtaTE(mintTEFee, { amount: tokenBalance });
+
+    whirlpools["Token-Token"] = await setupWhirlpool(mintA, mintB, tickSpacing);
+    whirlpools["Token-TE Token"] = await setupWhirlpool(
+      mintA,
+      mintTEA,
+      tickSpacing,
+    );
+    whirlpools["TE Token-TE Token"] = await setupWhirlpool(
+      mintTEA,
+      mintTEB,
+      tickSpacing,
+    );
+    whirlpools["Token-TE Token with Transfer Fee extension"] =
+      await setupWhirlpool(mintA, mintTEFee, tickSpacing);
+
+    positions["Token-Token"] = [
+      await setupPosition(whirlpools["Token-Token"]),
+      await setupPosition(whirlpools["Token-Token"], {
+        tickLower: 100,
+        tickUpper: 200,
+      }),
+      await setupTEPosition(whirlpools["Token-Token"]),
+      await setupTEPosition(whirlpools["Token-Token"], {
+        tickLower: 100,
+        tickUpper: 200,
+      }),
+    ];
+
+    positions["Token-TE Token"] = [
+      await setupPosition(whirlpools["Token-TE Token"]),
+      await setupPosition(whirlpools["Token-TE Token"], {
+        tickLower: 100,
+        tickUpper: 200,
+      }),
+      await setupTEPosition(whirlpools["Token-TE Token"]),
+      await setupTEPosition(whirlpools["Token-TE Token"], {
+        tickLower: 100,
+        tickUpper: 200,
+      }),
+    ];
+
+    positions["TE Token-TE Token"] = [
+      await setupPosition(whirlpools["TE Token-TE Token"]),
+      await setupPosition(whirlpools["TE Token-TE Token"], {
+        tickLower: 100,
+        tickUpper: 200,
+      }),
+      await setupTEPosition(whirlpools["TE Token-TE Token"]),
+      await setupTEPosition(whirlpools["TE Token-TE Token"], {
+        tickLower: 100,
+        tickUpper: 200,
+      }),
+    ];
+
+    positions["Token-TE Token with Transfer Fee extension"] = [
+      await setupPosition(
+        whirlpools["Token-TE Token with Transfer Fee extension"],
+      ),
+      await setupPosition(
+        whirlpools["Token-TE Token with Transfer Fee extension"],
+        {
+          tickLower: 100,
+          tickUpper: 200,
+        },
+      ),
+      await setupTEPosition(
+        whirlpools["Token-TE Token with Transfer Fee extension"],
+      ),
+      await setupTEPosition(
+        whirlpools["Token-TE Token with Transfer Fee extension"],
+        {
+          tickLower: 100,
+          tickUpper: 200,
+        },
+      ),
+    ];
+  });
+
+  const testLiquidityIncrease = async (
+    positionMint: Address,
```

Comment:
idea: since this actually only needs the positionMint (tokenA/tokenB address we can derive from the account data) we could potentially do the `for (mint in PositionMints) { it("abcee") }` approach.

For example in the before all if you do something like this:

```ts
const positions: Map<string, Address> = new Map(); // key is a human-readable name

beforeAll(async () => {
  const mintA = ...
  const mintB = ...
  // etc.
  
  const mintPairs = [
    [mintA, mintB]
    // etc.
  ]

  const pools = mintPairs.map(x => /* init pool */);

  const positionTypes = [
    { lowerTick: -100, upperTick: 100 },
    { lowerTick: -100, upperTick: -1 },
    { lowerTick: 1, upperTick: 100 }
  ] 

  for (const [name, pool] in pools) {
    for (const positionType in positionTypes) {
      positions.set(`${name} #{positionType}`, /* setupPosition */) 
    }
  }
})

for (const [name, positionMint] of positions.entries()) {
  it(`Increase liquidity ${name}`, () => {
    // write the test flow here
  })
}

// add the other tests
```